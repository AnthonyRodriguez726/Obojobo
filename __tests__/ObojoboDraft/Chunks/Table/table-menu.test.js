import React from 'react';
import { shallow, mount } from 'enzyme';
import TableMenu from '../../../../ObojoboDraft/Chunks/Table/table-menu.js';

// TODO: Test functionality of the onclick method
//       Test renderRow/renderCol functionality
//       Shallow render the component and snapshot test.

describe('TableMenu', () => {
  let onMenuCommandMock
  let tableMenu
  beforeEach(() => {
    onMenuCommandMock = jest.fn()
    tableMenu = (
      <TableMenu 
        type='row' 
        onMenuCommand={ onMenuCommandMock } 
        row={0} 
        col={0}
      />
    )
  })

  it('renders without blowing up', () => {
    expect(shallow(tableMenu)).toMatchSnapshot()
  })

  it('inserts a row above', () => {
    const wrapper = mount(tableMenu)
    wrapper.find('.insert-above').simulate('click')
    expect(onMenuCommandMock).toBeCalledWith({ row: 0, col: 0, command: 'insertRowAbove' })
  })

  it('inserts a row below', () => {
    const wrapper = mount(tableMenu)
    wrapper.find('.insert-below').simulate('click')
    expect(onMenuCommandMock).toBeCalledWith({ row: 0, col: 0, command: 'insertRowBelow' })
  })

  it('deletes a row', () => {
    const wrapper = mount(tableMenu)
    wrapper.find('li.delete').at(0).simulate('click')
    expect(onMenuCommandMock).toBeCalledWith({ row: 0, col: 0, command: 'deleteRow' })
  })
})

