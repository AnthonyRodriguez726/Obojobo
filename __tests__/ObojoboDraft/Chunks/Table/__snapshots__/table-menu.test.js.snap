// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableMenu renders without blowing up 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="obojobo-draft--chunks--table--table-menu row"
    style={
        Object {
            "backgroundImage": "",
          }
    }
>
    <ul>
        <li
            className="insert-above"
            onClick={[Function]}
        >
            Insert 1 row above
        </li>
        <li
            className="insert-below"
            onClick={[Function]}
        >
            Insert 1 row below
        </li>
        <li
            className="delete"
            onClick={[Function]}
        >
            Delete this row
        </li>
    </ul>
</div>,
  "nodes": Array [
    <div
      className="obojobo-draft--chunks--table--table-menu row"
      style={
            Object {
                  "backgroundImage": "",
                }
      }
>
      <ul>
            <li
                  className="insert-above"
                  onClick={[Function]}
            >
                  Insert 1 row above
            </li>
            <li
                  className="insert-below"
                  onClick={[Function]}
            >
                  Insert 1 row below
            </li>
            <li
                  className="delete"
                  onClick={[Function]}
            >
                  Delete this row
            </li>
      </ul>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TableMenu
        col={0}
        onMenuCommand={[Function]}
        row={0}
        type="row"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TableMenu {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "col": 0,
          "onMenuCommand": [Function],
          "row": 0,
          "type": "row",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="obojobo-draft--chunks--table--table-menu row"
          style={
                    Object {
                              "backgroundImage": "",
                            }
          }
>
          <ul>
                    <li
                              className="insert-above"
                              onClick={[Function]}
                    >
                              Insert 1 row above
                    </li>
                    <li
                              className="insert-below"
                              onClick={[Function]}
                    >
                              Insert 1 row below
                    </li>
                    <li
                              className="delete"
                              onClick={[Function]}
                    >
                              Delete this row
                    </li>
          </ul>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="obojobo-draft--chunks--table--table-menu row"
          style={
                    Object {
                              "backgroundImage": "",
                            }
          }
>
          <ul>
                    <li
                              className="insert-above"
                              onClick={[Function]}
                    >
                              Insert 1 row above
                    </li>
                    <li
                              className="insert-below"
                              onClick={[Function]}
                    >
                              Insert 1 row below
                    </li>
                    <li
                              className="delete"
                              onClick={[Function]}
                    >
                              Delete this row
                    </li>
          </ul>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TableMenu
    col={0}
    onMenuCommand={[Function]}
    row={0}
    type="row"
/>,
}
`;
